src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1273 @@
+                                    cmdStatus = Status::Failure);
+
+    {
+        // If the device successfully accepts the request, the server will attempt to operate at the location(s)
+        // indicated by the entries of the NewLocation field, when requested to operate,
+        // the SelectLocationsResponse command SHALL have the Status field set to Success,
+        // and the SelectedLocations attribute SHALL be set to the value of the NewLocations field.
+        mDelegate->ClearSelectedLocations();     
+
+        if (!req.newLocations.IsNull())
+        {
+            auto locationIter = req.newLocations.Value().begin();
+            uint32_t dummyIter;
+            while (locationIter.Next()) 
+            {
+                mDelegate->AddSelectedLocation(locationIter.GetValue(), dummyIter); 
+            }
+        }
+
+        NotifySelectedLocationsChanged();
+    }
+
+
+exit:
+    response.status = locationStatus;
+
+    switch (locationStatus)
+    {
+    // If the Status field is set to UnsupportedLocation, the StatusText field SHALL be an empty string.
+    // If the Status field is set to DuplicatedLocations, the StatusText field SHALL be an empty string.
+    case SelectLocationsStatus::kUnsupportedLocation:
+    case SelectLocationsStatus::kDuplicatedLocations:
+        response.statusText = chip::CharSpan::fromCharString("");
+        break;
+
+    // If the Status field is set to Success, the StatusText field is optional.
+    case SelectLocationsStatus::kSuccess:
+        if (useLocationStatusText)
+        {
+            response.statusText = chip::CharSpan::fromCharString( locationStatusText);
+        }
+        break;
+
+    // If the Status field is not set to Success, or UnsupportedLocation, or DuplicatedLocations,
+    // the StatusText field SHALL include a vendor-defined error description
+    default:
+        response.statusText = chip::CharSpan::fromCharString( locationStatusText);
+        break;
+
+    } // end switch
+
+
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, cmdStatus);
+    ctx.mCommandHandler.AddResponse(ctx.mRequestPath, response);
+}
+
+
+void Instance::HandleSkipCurrentCmd(HandlerContext & ctx)
+{
+    ChipLogDetail(Zcl, "Location Location: HandleSkipCurrent");
+
+    SkipCurrentStatus skipStatus                  = SkipCurrentStatus::kSuccess;
+    char   skipStatusText[kMaxSizeStatusText + 1] = {'\0'};
+    Status cmdStatus                              = Status::Success;
+
+    // On receipt of this command the device SHALL respond with a SkipCurrentLocationResponse command. 
+    Commands::SkipCurrentResponse::Type response;
+
+    // InvalidLocationList | The SelectedLocations attribute is null.
+    // If the Status field is set to InvalidLocationList, the StatusText field SHALL be an empty string.
+    VerifyOrExit((mDelegate->GetNumberOfSelectedLocations() == 0), 
```

Comment:
Here we want to verify that the SelectedLocations attribute is not null.

```suggestion
    VerifyOrExit((mDelegate->GetNumberOfSelectedLocations() != 0), 
```